pubsub/gcloud/aio/pubsub/subscriber_client.py

```diff
@@ -15,12 +20,30 @@ def __init__(self, **kwargs) -> None:
     from google.api_core import exceptions
     from google.cloud import pubsub
     from google.cloud.pubsub_v1.subscriber.message import Message
-    from google.cloud.pubsub_v1.subscriber.scheduler import Scheduler
-    from google.cloud.pubsub_v1.types import FlowControl
 
+    from .subscriber_message import SubscriberMessage
     from .utils import convert_google_future_to_concurrent_future
 
 
+    class FlowControl():
+        def __init__(self,
+                     max_bytes: int = 100 * 1024 * 1024,
+                     max_messages: int = 1000,
+                     max_lease_duration: int = 1 * 60 * 60,
+                     max_duration_per_lease_extension: int = 0) -> None:
+            self.max_bytes = max_bytes
+            self.max_messages = max_messages
+            self.max_lease_duration = max_lease_duration
+            self.max_duration_per_lease_extension = (
+                max_duration_per_lease_extension)
+
+        def __getitem__(self, index: int):
```

Comment:
Is this a feature we're using and/or want to expose? This seems a bit instructable to me, eg. perhaps it's something which the Google FlowControl construct accidentally exposes rather than an intentional part of the API?

Fine to keep it in as a transition, if we need it to help migrate towards doing this ourselves, but otherwise probably best to stick with a more clear access-by-name interface.

Reply:
Hah, sorry -- this is to obey the python - collections - sequence - interface _just enough_ that all expectations for what constitutes a tuple can be satisfied.